### PR TITLE
Multiple addresses for Client

### PR DESCRIPTION
**Pull request by ddustin**
Allow multiple addresses for client, falling back to the next one in case of failure.

@cpacia

### CODE REVIEW

File: obelisk/client.py [hunk 1]
```diff
@@ -64,9 +64,12 @@ class LibbitcoinClient(ClientBase):
         'validate'
     ]
 
-    def __init__(self, address, public_key=None, log=None, heartbeat_port=9092):
+    def __init__(self, address, addresses=None, public_key=None, log=None, heartbeat_port=9092):
```
Comment (cpacia) [inline obelisk/client.py]: maybe here we could combine these two arguments into one that accepts either a list or string.

And check it like:

``` python
if isinstance(address, list):
    self.address = address[0]
    self.address = address
else:
    self.address = address

Also we need to think of a good way select the initial url. Probably should just be random for now, but maybe we could do it by geographic location relative to our IP in the future. 

```

File: obelisk/client.py [hunk 2]
```diff
@@ -74,6 +77,16 @@ def __init__(self, address, public_key=None, log=None, heartbeat_port=9092):
...
+                index = self.addresses.index(self.address)
+                index = (index + 1) % len(self.addresses)
+                self.address = self.addresses[index]
+            except ValueError:
+                pass
+        return ClientBase._reconnect(self)
+
     def listen_heartbeat(self, port):
         def timeout():
             self.connected = False
```
Comment (cpacia) [inline obelisk/client.py:96]: We also should change servers if the hearbeat times out. The _reconnect call only times out on individual calls, but we might know before then if the server went down via the heartbeat.

File: obelisk/client.py (hunk 1 above)
Comment (ddustin) [inline obelisk/client.py]: I switched it over to allow address to be an array. I also randomized the server list over here:
https://github.com/OpenBazaar/OpenBazaar-Server/pull/303/commits/7cd19e00beb3ee7aaffd3ecc8f08345a6be06692

For geo ip targeting what service should we use? freegeoip.net looks good to me.

File: obelisk/client.py (hunk 2 above)
Comment (ddustin) [inline obelisk/client.py:96]: The commit https://github.com/OpenBazaar/python-libbitcoinclient/pull/3/commits/be16b5301796a99e6abac2bf97a086772c71e206 now does that.

Is there ever a scenario where _reconnect occurs and a heartbeat timeout? It wouldn't be ideal if it rotated servers twice.

**Pull request merged**